inc/saiuburst.h

```diff
@@ -0,0 +1,403 @@
+     * 
+     * @type sai_u32_t
+     * @flags  CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_HISTOGRAM_ATTR_RESOLUTION,
+
+    /**
+     * @brief Histogram Clear-On-Read Mode
+     *
+     * If true the histogram bin values are clear on read/transport.
+     * Otherwise values in bins are counting continuasly
+     * 
+     * @type boolean
+     * @flags  CREATE_AND_SET
+     * @default allownull true
+     */
+    SAI_TAM_HISTOGRAM_ATTR_CLEAR_MODE,
+
+
+    /**
+     * @brief Transporter Object
+     *
+     * Provides the transporter object for this histogram. When the 
+     * data shot is made, this transporter will be used to 'copy' 
+     * the data to the 'transporter-desired' location. In the 
+     * absence of a transporter, the tracker's default transporter 
+     * will be used (DEFAULT). 
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_TRANSPORTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_TAM_HISTOGRAM_ATTR_TRANSPORTER,
+
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_TAM_HISTOGRAM_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TAM_HISTOGRAM_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TAM_HISTOGRAM_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_histogram_attr_t;
+
+
+/**
+* Routine Description:
+*   @brief Create and return a microburst object
+*
+*   This creates a microburst in the hardware with the associated statistic
+*
+* Arguments:
+*   @param[out] microburst_id  - microburst object
+*   @param[in] attr_list � preferences for creating a microburst
+*
+* Return Values:
+*   @return SAI_STATUS_SUCCESS on success
+*          Failure status code on error
+*/
+typedef sai_status_t(*sai_create_tam_microburst_fn) (
+	_Out_ sai_object_id_t * microburst_id,
+	_In_ uint32_t attr_count,
+	_In_ sai_attribute_t * attr_list /* sai_tam_microburst_attr_t */
+	);
+
+/**
+* Routine Description:
+*   @brief Deletes a specified microburst object.
+*
+* Arguments:
+*   @param[in] microburst_id  - microburst object to be removed.
+*
+* Return Values:
+*    @return SAI_STATUS_SUCCESS on success
+*           Failure status code on error
+*/
+typedef sai_status_t(*sai_remove_tam_microburst_fn) (
+	_In_ sai_object_id_t * microburst_id
+	);
+
+/**
+* Routine Description:
+*   @brief Get values for specified microburst attributes.
+*
+* Arguments:
+*   @param[in] microburst_id � microburst object id
+*   @param[in] attr_count - number of attributes
+*   @param[inout] attr_list - array of attributes
+*
+* Return Values:
+*   @return SAI_STATUS_SUCCESS on success
+*           Failure status code on error
+*/
+typedef sai_status_t(*sai_get_tam_microburst_attribute_fn) (
+	_In_ sai_object_id_t microburst_id,
+	_In_ uint32_t attr_count,
+	_Inout_ sai_attribute_t * attr_list
+	);
+
+/**
+* Routine Description:
+*   @brief Set microburst attribute value(s).
+*
+* Arguments:
```

Comment:
Routine Description: Arguments:Return Values: are not needed doxygen produces them aumaticaly in documentation and they are not present in any otehgr header